op-supervisor/supervisor/backend/cross/safe_frontier.go

```diff
@@ -26,6 +26,10 @@ func HazardSafeFrontierChecks(d SafeFrontierCheckDeps, inL1Source eth.BlockID, h
 				// If not in cross-safe scope, then check if it's the candidate cross-safe block.
 				candidate, err := d.CandidateCrossSafe(hazardChainID)
 				if err != nil {
+					// ErrOutOfScope should be translated to an ErrFuture, since the dependency being out of scope does not warrant a Scope Bump of this chain.
+					if errors.Is(err, types.ErrOutOfScope) {
+						return fmt.Errorf("hazard dependency %s (chain %s) is out of scope: %w", hazardBlock, hazardChainID, types.ErrFuture)
+					}
 					return fmt.Errorf("failed to determine cross-safe candidate block of hazard dependency %s (chain %s): %w", hazardBlock, hazardChainID, err)
 				}
```

Comment:
```suggestion
				// ErrOutOfScope should be translated to an ErrFuture, since the dependency being out of scope does not warrant a Scope Bump of this chain.
				if errors.Is(err, types.ErrOutOfScope) {
					return fmt.Errorf("hazard dependency %s (chain %s) is out of scope: %w", hazardBlock, hazardChainID, types.ErrFuture)
				} else if err != nil {
					return fmt.Errorf("failed to determine cross-safe candidate block of hazard dependency %s (chain %s): %w", hazardBlock, hazardChainID, err)
				}
```
less indentation is easier to follow and it's more common to treat special error cases first, then have the remaining unspecific error case.